src/tracer/tracee_memory.rs

```diff
@@ -3,20 +3,37 @@ use libc::{c_ulonglong, c_void};
 use nix::sys::ptrace;
 use nix::unistd::Pid;
 
-fn cast_to_byte_array(word: c_ulonglong) -> [u8; 8] {
-    let ptr: &[u8; 8];
-    unsafe {
-        ptr = &*(&word as *const u64 as *const [u8; 8]);
-    }
-    *ptr
+fn cast_to_four_byte_array(x: i32) -> [u8; 4] {
+    [
+        (x & 0xff) as u8,
+        ((x >> 8) & 0xff) as u8,
+        ((x >> 16) & 0xff) as u8,
+        ((x >> 24) & 0xff) as u8,
+    ]
 }
 
-fn cast_to_word(bytes: [u8; 8]) -> c_ulonglong {
-    let void_ptr;
-    unsafe {
-        void_ptr = std::mem::transmute(bytes);
-    }
-    void_ptr
+fn cast_to_eight_byte_array(x: c_ulonglong) -> [u8; 8] {
+    [
+        (x & 0xff) as u8,
+        ((x >> 8) & 0xff) as u8,
+        ((x >> 16) & 0xff) as u8,
+        ((x >> 24) & 0xff) as u8,
+        ((x >> 32) & 0xff) as u8,
+        ((x >> 40) & 0xff) as u8,
+        ((x >> 48) & 0xff) as u8,
+        ((x >> 56) & 0xff) as u8,
+    ]
+}
+
+fn cast_to_eight_byte_word(bytes: [u8; 8]) -> c_ulonglong {
+    u64::from(bytes[0])
+        + (u64::from(bytes[1]) << 8)
+        + (u64::from(bytes[2]) << 16)
+        + (u64::from(bytes[3]) << 24)
+        + (u64::from(bytes[4]) << 32)
+        + (u64::from(bytes[5]) << 40)
+        + (u64::from(bytes[6]) << 48)
+        + (u64::from(bytes[7]) << 56)
 }
```

Comment:
I'm not sure I like this bit-shifting better, but -- as long as we have good test coverage -- I also don't care that much.

Reply:
i don't feel that strongly, i just thought it would be good to get rid of usage of `unsafe`. let me know if you'd like me to switch back. 

just to be clear, are you asking for more unit tests around these functions?

Reply:
No, the existing tests are fine. I'm just saying that I don't care that strongly and it's up to you if you want to switch back.